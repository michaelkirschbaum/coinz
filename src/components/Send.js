/** @jsx jsx */
import React from 'react'
import { css, jsx } from '@emotion/core'

class Send = () => (
  <div
    css={css`
      display: grid;
    `}
  >
    <div>Send Jobcoin</div>
    <div>
      Destination Address
      <input />
    </div>
    <div>
      Amount
      <input />
    </div>
    <button>Send</button>
  </div>
)

export default Send