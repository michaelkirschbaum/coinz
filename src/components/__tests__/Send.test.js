import React from 'react'
import { shallow } from 'enzyme/build';
import Send from 'components/Send';

describe('Send', () => {
  it('should match the snapshot', () => {
    const send = shallow(
      <Send />,
    )
    expect(send).toMatchSnapshot()
  })

  it('should display error if send address is invalid', () => {
    expect(false).toBe(true)
  })

  it('should display error if jobcoin amount is invalid', () => {
    expect(false).toBe(true)
  })

  it('should display error if insufficient funds for amount', () => {
    expect(false).toBe(true)
  })

  it('should display error if unable to retrieve balance', () => {
    expect(false).toBe(true)
  })

  it('show success when button clicked', () => {
    expect(false).toBe(true)
  })
})