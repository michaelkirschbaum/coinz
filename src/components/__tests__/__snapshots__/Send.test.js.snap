// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send should match the snapshot 1`] = `
<Send>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Send"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": undefined,
        "name": "1h62oil",
        "next": undefined,
        "styles": "
        padding: 0 10px 10px 10px;
        border: 1px solid lightgrey;
        overflow: hidden;
      ",
      }
    }
  >
    <div
      className="css-meg4kb-Send"
    >
      <h4>
        Send Jobcoin
      </h4>
      <form
        onSubmit={[Function]}
      >
        <label>
          <div>
            Destination Address
          </div>
          <EmotionCssPropInternal
            __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Send"
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="input"
            css={
              Object {
                "map": undefined,
                "name": "yk2vmo",
                "next": undefined,
                "styles": "width: 100%; box-sizing: border-box;",
              }
            }
            name="address"
            onChange={[Function]}
            type="text"
          >
            <input
              className="css-1o5t9mf-Send"
              name="address"
              onChange={[Function]}
              type="text"
            />
          </EmotionCssPropInternal>
        </label>
        <label>
          <div>
            Amount
          </div>
          <EmotionCssPropInternal
            __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Send"
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="input"
            css={
              Object {
                "map": undefined,
                "name": "yk2vmo",
                "next": undefined,
                "styles": "width: 100%; box-sizing: border-box;",
              }
            }
            name="amount"
            onChange={[Function]}
            type="text"
          >
            <input
              className="css-1o5t9mf-Send"
              name="amount"
              onChange={[Function]}
              type="text"
            />
          </EmotionCssPropInternal>
        </label>
        <EmotionCssPropInternal
          __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Send"
          __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
          css={
            Object {
              "map": undefined,
              "name": "khybjw",
              "next": undefined,
              "styles": "margin-top: 5px; float: right;",
            }
          }
        >
          <div
            className="css-sv49ul-Send"
          >
            <input
              type="submit"
              value="Send"
            />
          </div>
        </EmotionCssPropInternal>
      </form>
    </div>
  </EmotionCssPropInternal>
</Send>
`;
